conf.py

```diff
@@ -403,9 +405,25 @@ def localize(app, pagename, templatename, context, doctree):
         for la in app.config.languages.split(',')
         if la != current_lang
     ]
+    context['language_codes'] = [
+        (la.split('_')[0] if la != 'en' else 'x-default', _build_url(app.config.canonical_root, (la != 'en' and la or ''), pagename))
+        for la in app.config.languages.split(',')
+    ]
```

Comment:
Could be worth merging languages and languages code in a single traditional loop? Not really sure:
```python
context['languages'] = []
context['language_codes'] = []
for la in app.config.languages.split(','):
    url = _build_url(app.config.canonical_root, (la != 'en' and la or ''), pagename)
    context['langage_codes'].append((la.split('_')[0] if la != 'en' else 'x_default', url))
    if la != current_lang:
        context['langages'].append((LANGAGES.get(la, la.upper()), url)
```

Reply:
not sure two loops improve the readability IMHO, we only have a few items anyway